hippod/api_object_post.py

```diff
@@ -318,17 +346,20 @@ def try_adding_xobject(app, xobj):
     # object container file write should be delayed until everything
     # is sane.
     ok = is_obj_major_already_in_db(app, sha_major)
+    anchor_available = check_anchor(app, xobj)
+    # check whether anchor is in achievement to add correct initial lifetime of the object
     if not ok:
         # new entry, save to file
         # FULL update
-        hippod.store_container.save_new_object_container(app, sha_major, sha_minor, xobj,)
+        hippod.store_container.save_new_object_container(app, sha_major, sha_minor, xobj, anchor_available)
```

Comment:
why is check_anchor() not called within save_new_object_container()? the code should be as local as possible. Handing over a lot of arguments from outside make it hard to understand. For the human brain it is easier if time between generation (check_anchor()) and usage is near by each other.